src/pages/CommitDetailPage/CommitDetailPage.tsx

```diff
@@ -0,0 +1,124 @@
+import { useQueryClient } from '@tanstack/react-query'
+import { lazy, Suspense } from 'react'
+import { useParams } from 'react-router-dom'
+
+import NotFound from 'pages/NotFound'
+import { useFlags } from 'shared/featureFlags'
+import Breadcrumb from 'ui/Breadcrumb'
+import Spinner from 'ui/Spinner'
+import SummaryDropdown from 'ui/SummaryDropdown'
+
+import CommitBundleDropdown from './Dropdowns/CommitBundleDropdown'
+import CommitCoverageDropdown from './Dropdowns/CommitCoverageDropdown'
+import Header from './Header'
+import { useCommitPageData } from './hooks'
+
+const CommitCoverage = lazy(() => import('./CommitCoverage'))
+const CommitBundleAnalysis = lazy(() => import('./CommitBundleAnalysis'))
+
+const DISPLAY_MODE = {
+  COVERAGE: 'coverage',
+  BUNDLE_ANALYSIS: 'bundle-analysis',
+  BOTH: 'both',
+} as const
+
+type TDisplayMode = (typeof DISPLAY_MODE)[keyof typeof DISPLAY_MODE]
+
+const Loader = () => (
+  <div className="flex flex-1 justify-center">
+    <Spinner />
+  </div>
+)
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+  commit: string
+}
+
+const CommitDetailPage: React.FC = () => {
+  const { provider, owner, repo, commit: commitSha } = useParams<URLParams>()
+  const shortSHA = commitSha?.slice(0, 7)
+
+  const { bundleAnalysisPrAndCommitPages } = useFlags({
+    bundleAnalysisPrAndCommitPages: false,
+  })
+
+  // reset cache when user navigates to the commit detail page
+  const queryClient = useQueryClient()
+  queryClient.setQueryData(['IgnoredUploadIds'], [])
+
+  const { data: commitPageData, isLoading } = useCommitPageData({
+    provider,
+    owner,
+    repo,
+    commitId: commitSha,
+  })
+
+  if (
+    !isLoading &&
+    !commitPageData?.commit &&
+    !commitPageData?.isCurrentUserPartOfOrg
```

Comment:
qq: why do we care if user is not part of org?

Reply:
This is the current logic in the [commit detail page](https://github.com/codecov/gazebo/blob/main/src/pages/CommitDetailPage/CommitDetailPage.jsx#L35-L41), and also [untouched from when I restructured the commit detail page](https://github.com/codecov/gazebo/pull/2523/files#diff-dabc465f816518bc9044e32a22879443f18b1d4262edc6b9ca8968a84ea2f88fL65-L710